docs/pages/api.rst

```diff
@@ -221,14 +221,15 @@ When used as a ``Collection`` method, operations fall into a few main categories
 all
 ~~~
 
-Convert the collection into an array.
+Convert the collection into an array, re-indexing keys using ``Collection::normalize()`` to prevent data loss by default.
 
-.. warning:: This is a lossy operation because PHP array keys cannot be duplicated and must either be ``int`` or ``string``.
-            If you want to ensure no data is lost in the case of duplicate keys, look at the ``Collection::normalize()`` operation.
+.. warning:: An earlier version of this operation did not re-index keys by default, which meant at times data loss could occur.
+        The reason data loss could occur is because PHP array keys cannot be duplicated and must either be ``int`` or ``string``. 
+        The old behaviour can still be achieved by using the operation with the ``$normalize`` parameter as *false*.
```

Comment:
I think we should remove this parameter in the next major version and keep this behavior, WDYT ?

To restore the previous behavior, users can do: `iterator_to_array($collection->getIterator());`

Reply:
it's a bit trickier because of this: https://github.com/loophp/collection/issues/208#issuecomment-955522900